exchanges/futures/contract_test.go

```diff
@@ -0,0 +1,73 @@
+package futures
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestStringToContractSettlementType(t *testing.T) {
+	t.Parallel()
+	contractSettlementTypesMap := map[string]struct {
+		CT    ContractSettlementType
+		Error error
+	}{
+		"lInear":          {Linear, nil},
+		"LINEAR":          {Linear, nil},
+		"Inverse":         {Inverse, nil},
+		"unset":           {UnsetSettlementType, nil},
+		"hybRiD":          {Hybrid, nil},
+		"LinearOrInverse": {LinearOrInverse, nil},
+		"":                {UnsetSettlementType, nil},
+		"Quanto":          {Quanto, nil},
+		"QUANTO":          {Quanto, nil},
+		"Unknown":         {UnsetSettlementType, ErrInvalidContractSettlementType},
+	}
+	var val ContractSettlementType
+	var err error
+	for x := range contractSettlementTypesMap {
+		val, err = StringToContractSettlementType(x)
+		assert.Equalf(t, val, contractSettlementTypesMap[x].CT, "got %v, expected %v", val, contractSettlementTypesMap[x].CT)
+		assert.ErrorIs(t, err, contractSettlementTypesMap[x].Error)
+	}
+}
+
+func TestContractSettlementTypeString(t *testing.T) {
+	t.Parallel()
+	contractSettlementTypeToStringMap := map[ContractSettlementType]string{
+		UnsetSettlementType:         "unset",
+		Linear:                      "linear",
+		Inverse:                     "inverse",
+		Quanto:                      "quanto",
+		LinearOrInverse:             "linearOrInverse",
+		Hybrid:                      "hybrid",
+		ContractSettlementType(200): "unknown",
+	}
+	for k := range contractSettlementTypeToStringMap {
+		assert.Equal(t, k.String(), contractSettlementTypeToStringMap[k])
+	}
+}
+
+func TestContractTypeToString(t *testing.T) {
+	t.Parallel()
+	contractTypeToStringMap := map[ContractType]string{
+		Daily:             "day",
+		Perpetual:         "perpetual",
+		LongDated:         "long_dated",
+		Weekly:            "weekly",
+		Fortnightly:       "fortnightly",
+		ThreeWeekly:       "three-weekly",
+		Monthly:           "monthly",
+		Quarterly:         "quarterly",
+		SemiAnnually:      "semi-annually",
+		HalfYearly:        "half-yearly",
+		NineMonthly:       "nine-monthly",
+		Yearly:            "yearly",
+		Unknown:           "unknown",
+		UnsetContractType: "unset",
+		ContractType(200): "unset",
+	}
+	for k := range contractTypeToStringMap {
+		assert.Equal(t, k.String(), contractTypeToStringMap[k])
```

Comment:
```suggestion
	for k, v := range contractTypeToStringMap {
		assert.Equal(t, k.String(), v)
	}
```

Reply:
Pardon me 😢 for leaving a 👍 _react_ without applying the change in this case. I thought I did so ...